parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -1422,16 +1428,35 @@ public <D> CloseableIterable<D> build() {
         }
 
         if (batchedReaderFunc != null) {
-          return new VectorizedParquetReader<>(
-              file,
-              schema,
-              options,
-              batchedReaderFunc,
-              mapping,
-              filter,
-              reuseContainers,
-              caseSensitive,
-              maxRecordsPerBatch);
+          if (isComet) {
+            LOG.info("Comet vectorized reader enabled");
+            return new CometVectorizedParquetReader<>(
```

Comment:
@pvary I’ve seen your work on the reader/writer API proposal, but I haven’t kept up with all the details. We’re adding a `CometVectorizedParquetReader`, does this align with your proposal?